python-sdk/nuscenes/utils/splits.py

```diff
@@ -213,6 +215,52 @@ def create_splits_scenes(verbose: bool = False) -> Dict[str, List[str]]:
     return scene_splits
 
 
+def get_scenes_of_split(split_name: str, nusc : NuScenes, verbose: bool = False) -> List[str]:
+    """
+    Returns the scenes in a given split.
+    :param split_name: The name of the split.
+    :param nusc: The NuScenes instance to know where to look up potential custom splits.
+    :param verbose: Whether to print out statistics on a scene level.
+    :return: A list of scenes in that split.
+    """
+
+    if is_predefined_split(split_name=split_name):
+        return create_splits_scenes(verbose=verbose)[split_name]
+    else:
+        return get_scenes_of_custom_split(split_name=split_name, nusc=nusc)
+
+def is_predefined_split(split_name: str) -> bool:
+    """
+    Returns whether the split name is one of the predefined splits in the nuScenes dataset.
+    :param split_name: The name of the split.
+    :return: Whether the split is predefined.
+    """
+    return split_name in create_splits_scenes().keys()
+
+
+def get_scenes_of_custom_split(split_name: str, nusc : NuScenes) -> List[str]:
+    """Returns the scene names from a custom `splits.json` file, or None if the custom split does not exist."""
+
+    splits_file_path: str = _get_custom_splits_file_path(nusc)
+    if (not os.path.exists(splits_file_path)) or (not os.path.isfile(splits_file_path)):
+        raise ValueError(f"Custom split {split_name} requested, but no valid file found at {splits_file_path}.")
+
+    with open(splits_file_path, 'r') as file:
+        splits_data : dict = json.load(file)
+        if split_name not in splits_data.keys():
```

Comment:
Can we shift the chunk of code from this line all the way to the `return` statement out of the context manager (since we only need the context manager for loading the json)?

Reply:
Pulled it out of there now